src/jobflow/managers/fireworks.py

```diff
@@ -151,7 +155,14 @@ def run_task(self, fw_spec):
 
         if store is None:
             store = SETTINGS.JOB_STORE
-        store.connect()
+        if os.environ["FW_DATASERVER_PORT"]:
```

Comment:
This will trigger `KeyError` if `FW_DATASERVER_PORT` is not set.

```suggestion
        if "FW_DATASERVER_PORT" in os.environ:
```

Reply:
Ah yes, good point. Thanks!